Documentation/leak-detection.md

```diff
@@ -19,6 +19,8 @@ Before the build, the MarkAndCatalogFiles task runs.  This does a few things:
 
 There's no change in source-build operation in poisoning mode during the build.
 
+**Note**: During the build of the source-build-reference-packages repository (regardless of poisoning mode), reference packages have the ```System.Reflection.AssemblyMetadataAttribute("source", "source-build-reference-packages")``` attribute injected into their respective reference assemblies. This attribute is looked for after the build as part of the poisoning mode.
+
 ## After the build
 
-After the build, the CheckForPoison task is run on the source-build output directory.  It again unpacks any archives and packages recursively, and checks for all three kinds of markers injected before the build.  It then writes out a [report](poison-report-format.md) that details everything that was found.
+After the build, the CheckForPoison task is run on the source-build output directory.  It again unpacks any archives and packages recursively, and checks for the three kinds of markers injected before the build and the source-build-reference-packages attribute added during the build.  It then writes out a [report](poison-report-format.md) that details everything that was found.
```

Comment:
Readers may need to be reminded what exactly the three kinds of markers are. They're listed earlier in the doc but only subtly. It may be useful to call them out more explicitly.

Reply:
Good point! I've clarified the types of markers in https://github.com/dotnet/source-build/pull/3744/commits/bd5887ad4301b08945f4509837df257275985f5b